awkward/array/jagged.py

```diff
@@ -406,6 +407,56 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
             return None
         else:
             return JaggedArray(starts, stops, result)
+        
+
+    def argproduct(self, other):
+        import awkward.array.table 
+        if not isinstance(other, JaggedArray):
+            raise ValueError("array given isn't instance of JaggedArray; need JaggedArrays to proceed")
+        
+        if (len(self._starts) != len(other)):
+            raise ValueError("Number of events in each array must be equal")
+        
+        starts1 = self._starts
+        stops1 = self._stops
+        counts1 = stops1 - starts1
+
+        starts2 = other.starts
+        stops2 = other.stops
+        counts2 = stops2 - starts2
+        
+
+        pairs_counts = numpy.zeros(len(starts1)+1, dtype=self.INDEXTYPE)
+        pairs_counts[1:] = numpy.cumsum(counts1*counts2, dtype=self.INDEXTYPE)
+
+        def parents_from_offsets(offsets, content):
```

Comment:
content is needed in this function only to get the length of the output array. This is almost encoded into offsets. I wonder if a simpler solution could be used? Eg, to either pass in len(content) or to append the N+1 offset of the offsets array (which may be useful for other reasons)

Reply:
(actually my second suggestion looks to be already the case - eg, can I not do

out=numpy.full(offsets[-1],-1, dtype=selfINDEXTYPE)

Reply:
Yeah you are right! We need the length only. I will do it by tomorrow morning ( travelling now).

Reply:
Done!